### PR TITLE
refactor(nodebuilder): add archival flag 

### PR DESCRIPTION
**Pull request by walldiss**
Add warning log in preparation of depreciation of experimental-pruning flag

### CODE REVIEW

**Review by cristaloleg — DISMISSED** [review]
SGTM

File: nodebuilder/pruner/module.go [hunk 1]
```diff
@@ -36,6 +39,11 @@ func ConstructModule(tp node.Type, cfg *Config) fx.Option {
 		fx.Invoke(func(_ *pruner.Service) {}),
 	)
 
+	if cfg.EnableService && (tp == node.Full || tp == node.Bridge) {
+		log.Warnf("WARNING: `--experimental-pruning` flag will be removed in an upcoming release. " +
+			"Pruning will become the default mode for all nodes. " +
+			"If you want to retain history beyond the sampling window, please pass `--archival` flag.")
```
Comment (Wondertan) [inline nodebuilder/pruner/module.go:45]: This makes it sound like `--archival` is available today.

**Review by renaynay — COMMENTED** [review]
Agree with Hlib, let's specify that and also add it to the flag description

File: nodebuilder/pruner/module.go (hunk 1 above)
Comment (walldiss) [reply nodebuilder/pruner/module.go]: yeah, reworked

**Review by Wondertan — APPROVED** [review]
uTACK

**Review by cristaloleg — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
Thank you @walldiss

**Pull request merged**